app/components/works/file_row_component.rb

```diff
@@ -22,12 +22,18 @@ def filesize
     end
 
     def date_uploaded
-      return unless uploaded?
+      return unless saved?
 
       form.object.model.created_at
     end
 
+    # file has been uploaded but may not yet have been saved to the database model (and may not validate)
     def uploaded?
+      !form.object.try(:model)&.blob.nil?
+    end
+
+    # file has been uploaded and saved to the model
+    def saved?
       form.object.persisted?
     end
```

Comment:
distinguishing between a persisted file and one that has been uploaded (and has a blob) but is not yet saved to the database... that test for `uploaded?` is a bit unusual -- but without digging into the depths of Reform, was the best I could come up with that worked in all cases